File: Algorithms/01.Recursion/src/RecursiveDrawing.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class RecursiveDrawing {
    public static void main(String[] args) throws IOException {
        BufferedReader console = new BufferedReader(
                new InputStreamReader(System.in));

        int number = Integer.parseInt(console.readLine());

        drawing(number);
    }
    static void drawing(int number){
        if(number <= 0){
            return;
        }

        for(int j = 0; j < number; j++){
            System.out.print("*");
        }
        System.out.println();

        drawing(number - 1);

        for(int j = 0; j < number; j++){
            System.out.print("#");
        }
        System.out.println();
    }
}

File: Algorithms/01.Recursion/Lab/03.RecursiveDrawing/RecursiveDrawing.cs
using System;

namespace Drawing
{
    public class RecursiveDrawing
    {
        public static void Main()
        {
            int n = int.Parse(Console.ReadLine());
            Drawing(n);
        }

        private static void Drawing(int n)
        {
            if (n == 0)
            {
                return;
            }

            Console.WriteLine(new string('*', n));
            Drawing(n - 1);
            Console.WriteLine(new string('#', n));
        }
    }
}
